optionals/optional.go

```diff
@@ -0,0 +1,76 @@
+package optionals
+
+import "fmt"
+
+type Optional[T any] struct {
+	value *T
+}
+
+func Some[T any](t T) Optional[T] {
+	return Optional[T]{
+		value: &t,
+	}
+}
+
+func None[T any]() Optional[T] {
+	return Optional[T]{}
+}
+
+func (opt Optional[T]) IsSome() bool {
+	return opt.value != nil
+}
+
+func (opt Optional[T]) IsNone() bool {
+	return opt.value == nil
+}
+
+func (opt Optional[T]) Get() (T, error) {
```

Comment:
Would it be better to follow map lookup conventions and return `(T, bool)`?  That would allow something like
```
if v, ok := opt.Get(); ok {
  // ...
}
```

Reply:
Good idea. In b507211.